Realm/RLMSyncConfiguration.mm

```diff
@@ -270,32 +286,44 @@ - (BOOL)enableFlexibleSync {
     return make_sync_error(errorClass, @(error.message.c_str()), error.error_code.value(), custom);
 }
 
-static void setDefaults(SyncConfig& config, RLMUser *user) {
-    config.client_resync_mode = ClientResyncMode::Manual;
-    config.stop_policy = SyncSessionStopPolicy::AfterChangesUploaded;
-
+- (void)assignConfigErrorHandler:(RLMUser *)user {
     RLMSyncManager *manager = [user.app syncManager];
     __weak RLMSyncManager *weakManager = manager;
-    config.error_handler = [weakManager](std::shared_ptr<SyncSession> errored_session, SyncError error) {
+    RLMSyncErrorReportingBlock resetHandler = self.manualClientResetHandler;
+    _config->error_handler = [weakManager, resetHandler](std::shared_ptr<SyncSession> errored_session, SyncError error) {
         RLMSyncErrorReportingBlock errorHandler;
         @autoreleasepool {
             errorHandler = weakManager.errorHandler;
         }
-        if (!errorHandler) {
-            return;
-        }
         NSError *nsError = RLMTranslateSyncError(std::move(error));
         if (!nsError) {
             return;
         }
+        if (!errorHandler && !resetHandler) {
+            return;
+        } else if (!errorHandler && nsError.code != RLMSyncErrorClientResetError) {
+            return;
+        }
+
         RLMSyncSession *session = [[RLMSyncSession alloc] initWithSyncSession:errored_session];
         dispatch_async(dispatch_get_main_queue(), ^{
             // Keep the SyncSession alive until the callback completes as
             // RLMSyncSession only holds a weak reference
             static_cast<void>(errored_session);
-            errorHandler(nsError, session);
+            if (resetHandler && nsError.code == RLMSyncErrorClientResetError) {
+                resetHandler(nsError, session);
+            } else {
+                errorHandler(nsError, session);
+            }
         });
     };
+};
+
+static void setDefaults(SyncConfig& config, RLMUser *user) {
+    config.client_resync_mode = ClientResyncMode::Recover;
+    config.stop_policy = SyncSessionStopPolicy::AfterChangesUploaded;
+
+    RLMSyncManager *manager = [user.app syncManager];
```

Comment:
Appears to be unused?

Reply:
Used for the manager.authorizationHeaderName and manager.customRequestHeaders in 336 and 339.